src/store/reducers/pdisk/utils.ts

```diff
@@ -0,0 +1,54 @@
+import type {TEvPDiskStateResponse} from '../../../types/api/pdisk';
+import type {TStorageInfo} from '../../../types/api/storage';
+import type {TEvSystemStateResponse} from '../../../types/api/systemState';
+import {preparePDiskData} from '../../../utils/disks/prepareDisks';
+import {prepareNodeSystemState} from '../../../utils/nodes';
+import type {PreparedStorageGroup} from '../storage/types';
+import {prepareStorageGroupData} from '../storage/utils';
+
+export function preparePDiksDataResponse([pdiskResponse, nodeResponse]: [
+    TEvPDiskStateResponse,
+    TEvSystemStateResponse,
+]) {
+    const rawPDisk = pdiskResponse.PDiskStateInfo?.[0];
+    const preparedPDisk = preparePDiskData(rawPDisk);
+
+    const rawNode = nodeResponse.SystemStateInfo?.[0];
+    const preparedNode = prepareNodeSystemState(rawNode);
+
+    return {
+        ...preparedPDisk,
+        NodeId: preparedPDisk.NodeId ?? preparedNode.NodeId,
+        NodeHost: preparedNode.Host,
+        NodeType: preparedNode.Roles?.[0],
+        NodeDC: preparedNode.DC,
+    };
+}
+
+export function preparePDiskStorageResponse(
+    data: TStorageInfo,
+    pDiskId: number | string,
+    nodeId: number | string,
+) {
+    const preparedGroups: PreparedStorageGroup[] = [];
+
+    data.StoragePools?.forEach((pool) => {
```

Comment:
This function is rather difficult to read and needs a little refactoring as for me.